GitUI/ShellHelper.cs

```diff
@@ -0,0 +1,101 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using ConEmu.WinForms;
+using GitCommands;
+
+namespace GitUI
+{
+    public class ShellHelper
```

Comment:
Why do we need to add this functionality to `GitUI` project and not to `ConEmuWinForms`? 
I do think this functionality is related to `ConEmuControl` then anything else. If we were to add or replace ConEmu with a different console, we would likely need to provide implementations specific to the new console.

With that callsites would be simpler, e.g. instead of
```cs
ShellHelper.ChangeFolderInTerminal(_terminal, path);
```
will be:
```cs
_terminal.ChangeWorkingDirectory(path);
```

Reply:
Isn't `ConEmuWinForms` an external submodule, part of the ConEmu app? 

For the methods that take a `ConEmuControl` param, I can certainly turn them into extension methods to simplify the call, if you like.

Reply:
> Isn't ConEmuWinForms an external submodule, part of the ConEmu app?

ConEmuWinForms is an external lib, conemu-inside can be adopted though.
But extension is probably preferred.

Reply:
Okay, done.

Reply:
ConEmu is a submodule, correct. However we have long diverged from its upstream and no longer sync with it.
The new functionality is closely related to ConEmu, and it should reside in the submodule, and be exposed via an (ideally simple to use) API.
Should we decide to remove ConEmu, we should be able to remove the submodule.

Reply:
Ah, okay, got it about the status of ConEmu.

As for ShellHelper, half of the functionality in there is not directly related to ConEmu; it's more generic having to do with paths to shells. The methods that actually change the active folder are coupled to ConEmu functionality, but they're also conceptually related to the other shell methods in the class. And FormBrowse (and AppSettings) was already tightly coupled to ConEmu.

I can see the case for having some kind of facade class to use in FormBrowse instead of accessing ConEmu directly, but that's not how the existing code was written, and it's much bigger and more complicated than what this PR is trying to accomplish. My suggestion would be to deal with this PR on its own terms and add a task to the todo list to abstract away the specific console.